cardano_node_tests/utils/dbsync_check_tx.py

```diff
@@ -0,0 +1,537 @@
+
+from cardano_clusterlib import clusterlib
+
+from cardano_node_tests.utils import clusterlib_utils
+from cardano_node_tests.utils import dbsync_types
+from cardano_node_tests.utils import helpers
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _sum_mint_txouts(txouts: clusterlib.OptionalTxOuts) -> List[clusterlib.TxOut]:
+    """Calculate minting amount sum for records with the same token.
+
+    Remove address information - minting tokens doesn't include address, only amount and asset ID,
+    i.e. address information is not available in `ma_tx_mint` table.
+    Remove also datum hash, which is not available as well.
+    MA output is handled in Tx output checks.
+    """
+    mint_txouts: Dict[str, clusterlib.TxOut] = {}
+
+    for mt in txouts:
+        if mt.coin in mint_txouts:
+            mt_stored = mint_txouts[mt.coin]
+            mint_txouts[mt.coin] = mt_stored._replace(
+                address="", amount=mt_stored.amount + mt.amount, datum_hash=""
+            )
+        else:
+            mint_txouts[mt.coin] = mt._replace(address="", datum_hash="")
+
+    return list(mint_txouts.values())
+
+
+def _get_scripts_hashes(
+    cluster_obj: clusterlib.ClusterLib,
+    records: Union[clusterlib.OptionalScriptTxIn, clusterlib.OptionalMint],
+) -> Dict[str, Union[clusterlib.OptionalScriptTxIn, clusterlib.OptionalMint]]:
+    """Create a hash table of Tx Plutus data indexed by script hash."""
+    hashes_db: dict = {}
+
+    for r in records:
+        if not r.script_file:
+            continue
+        shash = cluster_obj.g_transaction.get_policyid(script_file=r.script_file)
+        shash_rec = hashes_db.get(shash)
+        if shash_rec is None:
+            hashes_db[shash] = [r]
+            continue
+        shash_rec.append(r)
+
+    return hashes_db
+
+
+def _db_redeemer_hashes(
+    records: List[dbsync_types.RedeemerRecord],
+) -> Dict[str, List[dbsync_types.RedeemerRecord]]:
+    """Create a hash table of redeemers indexed by script hash."""
+    hashes_db: dict = {}
+
+    for r in records:
+        shash = r.script_hash
+        shash_rec = hashes_db.get(shash)
+        if shash_rec is None:
+            hashes_db[shash] = [r]
+            continue
+        shash_rec.append(r)
+
+    return hashes_db
+
+
+def _compare_redeemer_value(
+    tx_rec: Union[clusterlib.ScriptTxIn, clusterlib.Mint], db_redeemer: dict
+) -> bool:
+    """Compare the value of the tx redeemer with the value stored on dbsync."""
+    if not (tx_rec.redeemer_file or tx_rec.redeemer_value):
+        return True
+
+    redeemer_value = None
+
+    if tx_rec.redeemer_file:
+        with open(tx_rec.redeemer_file, encoding="utf-8") as r:
+            redeemer_value = json.loads(r.read())
+    elif tx_rec.redeemer_value and db_redeemer.get("int"):
+        redeemer_value = {"int": int(tx_rec.redeemer_value)}
+    elif tx_rec.redeemer_value and db_redeemer.get("bytes"):
+        # we should ignore the first and last 2 chars because they represent
```

Comment:
(optional) since we are creating a new file maybe we can update the style of the comments at this moment